src/template.js

```diff
@@ -74,30 +73,34 @@ export const TemplateChunk = Object.freeze({
     // create the DOM if it wasn't created before
     this.createDOM(el)
 
-    if (this.dom) {
-      // create the new template dom fragment if it want already passed in via meta
-      this.fragment = fragment || this.dom.cloneNode(true)
-    }
+    // <template> tags require a bit more work
+    // the template fragment might be already created via meta outside of this call
+    // create the new template dom fragment if it want already passed in via meta
+    const fragment = meta.fragment || (this.dom ? this.dom.cloneNode(true) : null)
 
     // store root node
     // notice that for template tags the root note will be the parent tag
     this.el = this.isTemplateTag ? parentNode : el
     // create the children array only for the <template> fragments
-    this.children = this.isTemplateTag ? children || Array.from(this.fragment.childNodes) : null
+    this.children = this.isTemplateTag && (children || fragment) ? children || Array.from(fragment.childNodes) : null
 
     // inject the DOM into the el only if a fragment is available
-    if (!avoidDOMInjection && this.fragment) injectDOM(el, this.fragment)
+    if (!avoidDOMInjection) {
+      if (fragment) injectDOM(el, fragment)
+      else if (isTemplateTag) el.parentNode.removeChild(el)
+    }
 
     // create the bindings
     this.bindings = this.bindingsData.map(binding => createBinding(
       this.el,
       binding,
       templateTagOffset
-    ))
+    )).filter(binding => binding)
     this.bindings.forEach(b => b.mount(scope, parentScope))
 
     // store the template meta properties
     this.meta = meta
+    this.fragment = fragment
```

Comment:
Previously, template was written to this only if this.dom was present. In the absence of this.dom and the presence of meta.fragment, this did not happen. Now it is always recorded and perhaps this creates some kind of problem that I could not find.

Reply:
This is the most complex part of the framework and you are digging deep into the core. [Udomdiff](https://github.com/riot/dom-bindings/blob/main/src/util/udomdiff.js) plays a big role here

Reply:
I am not sure it's worth it due to the nature of the edge case that makes no sense.

Reply:
I agree with you about the edge case. It doesn't make sense to create empty `<template>` tags, but I ran into an exception during development because I was just creating the skeleton of the component and it was pretty frustrating to get an error in a syntactically correct (albeit meaningless) template.

Reply:
I don't think I broke anything, all tests pass and I tried completely different use cases. Everything looks reliable ..